api/model.go

```diff
@@ -120,6 +120,26 @@ type EngineUpgradeInput struct {
 	Image string `json:"image"`
 }
 
+type Node struct {
+	client.Resource
+	Name            string                `json:"name"`
+	AllowScheduling bool                  `json:"allowScheduling"`
+	Disks           map[string]types.Disk `json:"disks"`
+}
+
+type DiskInput struct {
```

Comment:
I think it's maybe unnecessary to separate API for add/delete disk.

Considering in the UI, normally it is not going to call API every time when a disk was added or removed. It's easy for UI to call backend that the things have been determined. Also it's easier for the backend to store it in one transaction. And it's a rarely modified parameter, performance shouldn't be a problem. So it should be OK to just use `UpdateNode` to perform the update on all the fields of the node.

So many code below can be simplied.